src/Services/Batch/AbstractBatchService.php

```diff
@@ -54,18 +54,31 @@ abstract class AbstractBatchService
 
     /**
      * @param VariationElasticSearchScrollRepositoryContract $variationElasticSearchScrollRepository
+     * @throws \Exception
      */
     public function __construct(VariationElasticSearchScrollRepositoryContract $variationElasticSearchScrollRepository)
     {
         $this->settingshelper = pluginApp(SettingsHelper::class);
 
         /** @var CatalogExportRepositoryContract $catalogExportRepository */
         $catalogExportRepository = pluginApp(CatalogExportRepositoryContract::class);
+        /** @var CatalogRepositoryContract $catalogRepository */
         $catalogRepository = pluginApp(CatalogRepositoryContract::class);
-        $catalogRepository->setFilters(['template' => self::TEMPLATE]);
+        $catalogRepository->setFilters([
+            'template' => self::TEMPLATE,
+            'active' => true
+        ]);
         $id = null;
         /** @var Collection $mappings */
         $mappings = $catalogRepository->all()->getResult();
+
+        if (count($mappings) > 1) {
+            // Currently due to lack of filtering and the low API limit we can not export multiple catalogs.
+            // Since we can't know which catalog the customer wants to export we abort at this point
+            // if multiple active catalogs with the etsy template exist
+            throw new \Exception('Multiple active catalogs set.');
```

Comment:
I feel your Bächtle way of life, but we should then have a super nice pretty awesome log.

Reply:
Totally true, I will change that later on